accounts/usbwallet/ledger.go

```diff
@@ -170,6 +172,24 @@ func (w *ledgerDriver) SignTx(path accounts.DerivationPath, tx *types.Transactio
 	return w.ledgerSign(path, tx, chainID)
 }
 
+// SignTypedMessage implements usbwallet.driver, sending the message to the Ledger and
+// waiting for the user to sign or deny the transaction.
+//
+// Note: this was introduced in the ledger 1.5.0 firmware
+func (w *ledgerDriver) SignTypedMessage(path accounts.DerivationPath, domainHash []byte, messageHash []byte) ([]byte, error) {
+	// If the Ethereum app doesn't run, abort
+	if w.offline() {
+		return nil, accounts.ErrWalletClosed
+	}
+	// Ensure the wallet is capable of signing the given transaction
+	if w.version[0] < 1 && w.version[1] < 5 {
+		//lint:ignore ST1005 brand name displayed on the console
+		return nil, fmt.Errorf("Ledger v%d.%d.%d doesn't support EIP-712 signing, please update to v1.5.0 at least", w.version[0], w.version[1], w.version[2])
```

Comment:
It's a bit contrary to common practice to have an error message say things like "Please update to... ". Perhaps better with just "Ledger version >= 1.5.0 required for EIP-712 signing (found version v%d.%d.%d)"

Reply:
I was following the pattern of https://github.com/ethereum/go-ethereum/blob/master/accounts/usbwallet/ledger.go#L167, but have changed to your version now